evm-ds/src/evm_server_run.rs

```diff
@@ -0,0 +1,330 @@
+    trap_data_call.set_call_data(interrupt.input.into());
+    trap_data_call.set_is_static(interrupt.is_static);
+    trap_data_call.set_target_gas(interrupt.target_gas.unwrap_or(u64::MAX));
+
+    let mut trap_data = EvmProto::TrapData::new();
+    trap_data.set_call(trap_data_call);
+    result.set_trap_data(trap_data);
+    result.set_continuation_id(cont_id);
+    result
+}
+
+fn build_crate_result(
+    runtime: &Runtime,
+    interrupt: CpsCreateInterrupt,
+    traces: Vec<String>,
+    remaining_gas: u64,
+    cont_id: u64,
+) -> EvmProto::EvmResult {
+    let mut result = EvmProto::EvmResult::new();
+
+    result.set_return_value(runtime.machine().return_value().into());
+    let mut trap_reason = EvmProto::ExitReason_Trap::new();
+    trap_reason.set_kind(EvmProto::ExitReason_Trap_Kind::CREATE);
+    let mut exit_reason = EvmProto::ExitReason::new();
+    exit_reason.set_trap(trap_reason);
+    result.set_exit_reason(exit_reason);
+    result.set_trace(traces.into_iter().map(Into::into).collect());
+    result.set_remaining_gas(remaining_gas);
+
+    let mut scheme = EvmProto::TrapData_Scheme::new();
+
+    match interrupt.scheme {
+        evm::CreateScheme::Legacy { caller } => {
+            let mut scheme_legacy = EvmProto::TrapData_Scheme_Legacy::new();
+            scheme_legacy.set_caller(caller.into());
+            scheme.set_legacy(scheme_legacy);
+        }
+        evm::CreateScheme::Create2 {
+            caller,
+            code_hash,
+            salt,
+        } => {
+            let mut scheme_crate2 = EvmProto::TrapData_Scheme_Create2::new();
+            scheme_crate2.set_caller(caller.into());
+            scheme_crate2.set_code_hash(code_hash.into());
+            scheme_crate2.set_salt(salt.into());
+            scheme_crate2.set_create2_address(interrupt.create2_address.into());
+            scheme.set_create2(scheme_crate2);
+        }
+        evm::CreateScheme::Fixed(address) => {
+            let mut scheme_fixed = EvmProto::TrapData_Scheme_Fixed::new();
+            scheme_fixed.set_addres(address.into());
+            scheme.set_fixed(scheme_fixed);
+        }
+    }
+    let mut trap_data_create = EvmProto::TrapData_Create::new();
+    trap_data_create.set_scheme(scheme);
+    trap_data_create.set_caller(interrupt.caller.into());
+    trap_data_create.set_call_data(interrupt.init_code.into());
+    trap_data_create.set_target_gas(interrupt.target_gas.unwrap_or(u64::MAX));
+    trap_data_create.set_value(interrupt.value.into());
+    let mut trap_data = EvmProto::TrapData::new();
+    trap_data.set_create(trap_data_create);
+    result.set_trap_data(trap_data);
+    result.set_continuation_id(cont_id);
+    result
+}
+
+fn handle_panic(traces: Vec<String>, remaining_gas: u64) -> EvmProto::EvmResult {
+    let mut result = EvmProto::EvmResult::new();
+    let mut fatal = EvmProto::ExitReason_Fatal::new();
+    fatal.set_kind(EvmProto::ExitReason_Fatal_Kind::OTHER);
```

Comment:
I think Fatal::Kind::OTHER in the protobuf assumes error_string should be set in the same proto.
Why not accept a string as an argument to this function, and pass it to provide error context?